activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -289,14 +289,14 @@ def enable_extension(name)
       # locks
       #
       # Return true if we got the lock, otherwise false
-      def get_advisory_lock(key) # :nodoc:
+      def get_advisory_lock(lock_id) # :nodoc:
```

Comment:
I followed the pattern set by the other 'template' methods.

We actually do have a method called `supports_advisory_locks?` that is implemented on abstract adapter and returns false there. Postgres and Mysql override this to return true.